app/stv/stv.js

```diff
@@ -0,0 +1,370 @@
+          action: 'WIN',
+          alternative,
+          voteCount: Number(voteCount.toFixed(4)),
+        });
+
+        // Find the done Votes
+        for (let i in votes) {
+          // @const { Vote } vote - The vote for this loop
+          const vote = JSON.parse(JSON.stringify(votes[i]));
+
+          // Votes that have the winning alternative as their first pick
+          if (vote.priorities[0]._id === alternative._id) doneVotes[i] = {};
+        }
+      }
+    }
+
+    // @let { [key: string]: {} } doneAlternatives - Have won or been eliminated
+    let doneAlternatives = {};
+
+    // @let { Vote[] } nextRoundVotes - The votes that will go on to the next round
+    let nextRoundVotes = [];
+
+    // If there are new winners this round
+    if (Object.keys(roundWinners).length > 0) {
+      // Check STV can terminate and return the RESOLVED winners
+      if (winners.length === seats) {
+        return {
+          result: { status: 'RESOLVED', winners },
+          log,
+          thr,
+          seats,
+          voteCount: inputVotes.length,
+          useStrict,
+        };
+      }
+
+      // Set the done alternatives as the roundwinners
+      doneAlternatives = roundWinners;
+
+      // The next rounds votes are votes that are not done.
+      nextRoundVotes = votes.filter((_, i) => !doneVotes[i]);
+
+      // Go through all done votes
+      for (let i in doneVotes) {
+        // @const { Vote } vote - The vote for this loop
+        const vote = JSON.parse(JSON.stringify(votes[i]));
+
+        // @const { Alternative } alternative - Take the first choice of the done vote
+        const alternative = JSON.parse(JSON.stringify(vote.priorities[0]));
+
+        // @const { float } fraction - Find the excess fraction for this alternative
+        const fraction = excessFractions[alternative._id] || 0;
+
+        // If the fraction is 0 (meaning no votes should be transferred) or if the vote
+        // has no more priorities (meaning it's exhausted) we can continue without transfer
+        if (fraction === 0 || vote.priorities.length === 1) continue;
+
+        // Fractional transfer. We mutate the weight for these votes by a fraction
+        vote['weight'] = vote.weight * fraction;
+        // Push the mutated votes to the list of votes to be processed in the next iteration
+        nextRoundVotes.push(vote);
+      }
+    } else {
+      // Find the lowest score
+      const minScore = Math.min(
+        ...alternatives.map(
+          (alternative) => counts[alternative.description] || 0
+        )
+      );
+
+      // Find the candidates with the lowest score
+      const minAlternatives = alternatives.filter(
+        (alternative) =>
+          (counts[alternative.description] || 0) <= minScore + EPSILON
+      );
+
+      // There is a tie for eliminating candidates. Per Scottish STV we must look at the previous rounds
+      if (minAlternatives.length > 1) {
+        let reverseIteration = iteration;
```

Comment:
This is also a bit confusing to me. We look at previous iterations to determine the loser, right? But here we set reverseIteration to the same as the current iteration. I'm assuming that this only makes it so that the first attempt at backwards tracing just fails (since it's the current one), and continues backward. Could we skip this step and check the previous iteration immediately? If not, and this is intentional, add a comment explaining why.

Reply:
Correct. I wrote the reverse algorithm as you describe above first but it resulted in more complex code. 

It need a lot of `in reverseIteration-1` there happend something, so we check `reverseIteration`, but first we must check that `reverseIteration` is not 0 because that means that `reverseIteration-1` actually was 1 and we should have stopped`.

This method _(that yeah sometimes runs one extra iterasion)_ was more understandable _(and cleaner)_ in my view. In this method the `while predicate` tells the algorithm when to stop, which i think makes sence.

Reply:
I see. But couldn't you set `reverseIteration = iteration - 1`. Then the while predicate would catch the `reverseIteration = 0` case and we skip the `iteration = reverseIteration` step?

I.e we just initialize `reverseIteration` "one lower".

And I see now that this must be `let` actually 😅

Reply:
I had that initially aka `reverseIteration = iteration - 1`, but then you need some extra (if) and you need to move and or duplicate the elimination code

Reply:
This while loop works if you are at iteraion 1 or N, and it will handle the logic the same either way. I promise, it's cleaner. :good code: ™️

Reply:
I think I understand now. I initially though it would make no difference to change this, but I see that the `MULTI_TIE_ELIMINATIONS` thing is the reason we cannot just ignore `iteration == reverseIteration`.  Since we actually don't always look at the last one, but we might end up just looking at the current iteration and eliminating all minAlternatives. 

You should change the comments/naming to reflect this.